openpype/plugins/publish/extract_jpeg_exr.py

```diff
@@ -62,100 +91,140 @@ def process(self, instance):
-                ))
-                continue
-
-            # Do conversion if needed
-            #   - change staging dir of source representation
-            #   - must be set back after output definitions processing
-            convert_dir = None
-            if do_convert:
-                convert_dir = get_transcode_temp_directory()
-                filename = os.path.basename(full_input_path)
-                convert_input_paths_for_ffmpeg(
-                    [full_input_path],
-                    convert_dir,
-                    self.log
-                )
-                full_input_path = os.path.join(convert_dir, filename)
-
+           
+            
+            # Presumably the following is not needed since we are being
+            # explicit
+            # TODO: Test cases, cleanup.
+            # do_convert = should_convert_for_ffmpeg(full_input_path)
+            # # If result is None the requirement of conversion can't be
+            # #   determined
+            # if do_convert is None:
+            #     self.log.info((
+            #         "Can't determine if representation requires conversion." # noqa
+            #         " Skipped."
+            #     ))
+            #     continue
+            #
+            # # Do conversion if needed
+            # #   - change staging dir of source representation
+            # #   - must be set back after output definitions processing
+            # convert_dir = None
+            # if do_convert:
+            #     convert_dir = get_transcode_temp_directory()
+            #     filename = os.path.basename(full_input_path)
+            #     convert_input_paths_for_ffmpeg(
+            #         [full_input_path],
+            #         convert_dir,
+            #         self.log
+            #     )
+            #     full_input_path = os.path.join(convert_dir, filename)
             filename = os.path.splitext(input_file)[0]
             if not filename.endswith('.'):
                 filename += "."
             jpeg_file = filename + "jpg"
             full_output_path = os.path.join(stagingdir, jpeg_file)
-
-            self.log.info("output {}".format(full_output_path))
-
-            ffmpeg_path = get_ffmpeg_tool_path("ffmpeg")
-            ffmpeg_args = self.ffmpeg_args or {}
-
-            jpeg_items = []
-            jpeg_items.append(path_to_subprocess_arg(ffmpeg_path))
-            # override file if already exists
-            jpeg_items.append("-y")
-            # use same input args like with mov
-            jpeg_items.extend(ffmpeg_args.get("input") or [])
-            # input file
-            jpeg_items.append("-i {}".format(
-                path_to_subprocess_arg(full_input_path)
-            ))
-            # output arguments from presets
-            jpeg_items.extend(ffmpeg_args.get("output") or [])
-
-            # If its a movie file, we just want one frame.
+            # If it's a movie file, use ffmpeg
             if repre["ext"] == "mov":
```

Comment:
- First of all, do not crash if oiio is not supported.
- Secondly, separate the extraction logic using ffmpeg and oiio into their separated methods. They should return if were successfull and should not crash if were not (thumbnail is not reason to crash publishing?). Failed extraction should cause warning log.
- Using extensions would be endless goal. Thus I recommend that you can check if oiio supports a file just by simply try to read information about it. If can't read their information then you can try convert it using ffmpeg.

```
from openpype.lib import oiio_is_supported, execute

thumbnail_created = False
# Try use ffmpeg when oiio is not supported (for cases when oiio tool is not available)
if not oiio_is_supported():
    thumbnail_created = self.create_thumbnail_with_ffmpeg(src_path, dst_path)
else:
    # Check if file can be read by oiio
    args = [
        oiiotool_path, "--info", "-i", src_path
    ]
    returncode = execute(args)
    
    # If input can be read by oiio then use oiio method for creation otherwise use ffmpeg
    if returncode == 0:
        thumbnail_created = self.create_thumbnail_with_ffmpeg(src_path, dst_path)
    else:
        thumbnail_created = self.create_thumbnail_oiio(src_path, dst_path)

# Skip rest of process if thumbnail was not created
if not thumbnail_created:
    return

# Rest of plugin logic
...
```